### ISSUE TITLE
CompositeCredential: make it possible to go with the token alone

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#4134]
Some times token is provided to the users, so we should allow for pure-token authentication.  ATM if there is no token, we go to the UserPassword, which is the first in the list of composite credentials.

May be while establishing connection we could just ask user, which of the components in composite credential they have?  token or user/password and let them choose.

**Comment by jbpoline** [comment]
@yarikoptic  Not sure if this is related or if this should be a new issue, Loris still has trouble : 
[](https://github.com/CONP-PCNO/conp-portal/issues/218)
Let me know if you see something obvious. Also, what key information should we have to see what is happening ?

**Comment by yarikoptic** [comment]
Just looked at it -- in https://github.com/CONP-PCNO/conp-portal/issues/218 they figured it was (way) too old of a datalad, so at least there is no rush with this issue ;)

**Comment by jbpoline** [comment]
Yes, was going to write that .. sorry for the false alarm. It was supposed to be "a fresh ubuntu install"  - not sure what version was that.  There is the annex version in the .git/config file, but does datalad stores its version somewhere ? (not in .datalad/config afaics)

**Comment by yarikoptic** [comment]
> "a fresh ubuntu install" 

sure thing, some 2 year old LTS (since it must be the long term! ;)), so it would come with 3 year old DataLad.

There is no "datalad version" to store since generally it shouldn't matter, but this example kinda gives a counter argument -- that there might be some cases that delivered `.datalad/` content could demand some minimal datalad version to operate on it.  We have something like that for handling metadata layouts, so if datalad detects some layout it doesn't yet know -- it would ask people to upgrade.  So, in this case, if ancient datalad had a clue even to look for providers configuration under `.datalad/` - it might have had a chance to end up spitting something like
```
$> datalad get 9931234/candidate.json
[WARNING] Running get resulted in stderr output: [ERROR] Unknown authentication_type=loris-token22222ZZg. Known are: html_form, http_auth, http_basic_auth, http_digest_auth, bearer_token, aws-s3, nda-s3, loris-token, xnat, none (ValueError) 
```
(I intentionally misconfigured to have some unknown to current datalad authentication_type) -- which would have suggested something, but it wouldn't know that such a type will come in the future within datalad.

Nevertheless!  Soon I will just submit a PR adding etelemetry which would announce that user is running old release.

**Comment by yarikoptic** [comment]
note: somewhat/might be related `HTTPAnonBearerTokenAuthenticator` (possibly to be introduced) in #5129 - a "proper" handling for bearertoken "authentication".

**Comment by jbpoline** [comment]
@yarikoptic thanks - seems very relevant - will foward that to @cecile @xavier @tom-gee (when I find their handles ...)
